### PR TITLE
Feature/as 3122 saving to db

### PR DESCRIPTION
**Pull request by shepda**
1. Update the api to save has appeal data with a stored procedure. This accepts a json string as a param, fetched the latest record, merges the json data with the latest record and saves as a new record. The trigger then makes the new record the latest record.

2. Update the appeals journey pages in the web-app to save data after each page submission. The json created after each page submission needs to have certain field names to match what the stored procedure is expecting and this mapping is handled by `src/config/db-fields.js`. Any additional fields in the json that the stored procedure isn't expecting will be ignored.

### CODE REVIEW

File: packages/api/database/migrations/00025-create-create-has-appeal-procedure.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        @inspectorFirstName = InspectorFirstName,
+        @inspectorSurname = InspectorSurname,
+        @inspectorId = InspectorID,
+        @inspectorSpecialismId = InspectorSpecialismID,
+        @scheduledSiteVisitDate = ScheduledSiteVisitDate,
+        @decisionOutcomeId = DecisionOutcomeID,
+        @decisionLetterId = DecisionLetterID,
+        @decisionDate = DecisionDate,
+        @descriptionDevelopment = DescriptionDevelopment
+      FROM HASAppeal
```
Comment (rcmsoftwaredevelopment) [inline packages/api/database/migrations/00025-create-create-has-appeal-procedure.js:59]: Can this be done with (nolock)

**Review by rcmsoftwaredevelopment — APPROVED** [review]

File: packages/api/database/migrations/00025-create-create-has-appeal-procedure.js (hunk 1 above)
Comment (shepda) [reply packages/api/database/migrations/00025-create-create-has-appeal-procedure.js:59]: Sure, will do

**Review by rcmsoftwaredevelopment — APPROVED** [review]

**Pull request merged**